add basename(payments) to payments router

Dogober opened the pull request.

Dogober committed:
add basename(payments) to payments router

Dogober committed:
Integrate PayPal Checkout

- Added PayPalPayment model inheriting from abstract Payment model.
- Implemented create_order method in PayPalPayment model to create PayPal orders.
- Created PayPalPaymentSerializer for validating and serializing PayPalPayment data.
- Added CreatePayPalPaymentView to handle PayPal payment creation requests.
- Added PayPalPaymentSuccessView to handle successful PayPal payment redirects.
- Added PayPalPaymentCancelView to handle canceled PayPal payment redirects.
- Updated exception handling in PayPalPayment model to use specific PayPal SDK exceptions.
- Configured URLs for PayPal payment creation and success/cancel redirects.
- Added client ID and secret key for PayPal to settings.py.
- Updated urls.py to include router and specific PayPal endpoints.

Dogober committed:
shered stripe and paypall data with others

Dogober committed:
Merge branch 'develop' into develop-payments-nick

Dogober committed:
Add unique related_name to borrowing field in Payment model

- Updated the borrowing field in the abstract Payment model to use related_name='%(class)s_payments' to ensure unique reverse query names for derived models.
- Removed redundant borrowing field definitions in StripePayment and PayPalPayment models.
- This change prevents reverse query name conflicts for borrowing field in derived models.

MindDevastation commented on src/payments/views.py:
```diff
@@ -43,7 +48,83 @@ def create(self, request, *args, **kwargs):
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        if serializer.is_valid():
+            payment, approval_url = serializer.save()
```
serializer.save() returns **payment and approval_url** but DRF expected only **one** argument
serializer.save() should return **payment** and approval_url should be transmited separately

MindDevastation commented on src/payments/views.py:
```diff
@@ -43,7 +48,83 @@ def create(self, request, *args, **kwargs):
+                paypal_payment = PayPalPayment.objects.get(paypal_order_id=token)
+                paypal_payment.payer_id = payer_id
+
+                order = paypalrestsdk.Order.find(token)
```
PayPal API dont have Order.create() use paypalrestsdk.Payment() instead

MindDevastation commented on src/payments/models.py:
```diff
@@ -78,3 +85,62 @@ def create_payment_intent(self):
+            )
+
+            if order.create():
+                self.paypal_order_id = order.id
```
paypalrestsdk.Payment() dont have .id. You should use payment.id

MindDevastation reviewed (changes requested):
Fix problems

MindDevastation commented on src/payments/urls.py:
```diff
@@ -1,11 +1,23 @@
 router = routers.DefaultRouter()
-router.register("payments", PaymentViewSet)
+router.register("payments", PaymentViewSet, basename="payments")
 router.register("stripe", StripePaymentViewSet)
```
add basename for stripe

Dogober committed:
Fix PayPal order creation using post method

- Updated PayPalPayment model to use post method for creating orders.
- Replaced non-existent create method with post for creating PayPal orders.
- Updated create_order method to handle response and extract approval_url.
- Modified CreatePayPalPaymentView to correctly handle PayPal order creation and return approval_url.

Dogober committed:
Merge branch 'develop' into develop-payments-nick

Dogober committed:
add PayPalModel and Serializer to PaymentViewSet

Dogober committed:
add basename for stripe

MindDevastation committed:
Merge branch 'develop' into develop-payments-nick
